pkg/registry/service/rest.go

```diff
@@ -342,6 +443,11 @@ func (rs *REST) Update(ctx api.Context, name string, objInfo rest.UpdatedObjectI
 		service.Status.LoadBalancer = api.LoadBalancerStatus{}
 	}
 
+	success, err := rs.healthCheckNodePortUpdate(oldService, service)
```

Comment:
Actually, Delete already has the feature gate. healthCheckNodePortUpdate() internally has the feature gate check, but it is not complete, will fix that up.